### PR TITLE
Myos 4 extended printf

### PR DESCRIPTION
**Pull request by BognaLew**
added options %d, %i, %x, %X, %p and %b (custom for printing binary numbers) to function printf. Also implemented functions itoa (convert integer to string), abs (absolute value), strrev (reverse string), strlwr (convert string to lowercase), strupr (convert string to uppercase).

### CODE REVIEW

**Review by bstrzelecki — APPROVED** [review]

File: libc/stdio/printf.c [hunk 1]
```diff
@@ -12,67 +13,143 @@ static bool print(const char* data, size_t length) {
...
+                        (*format) += len;
+			break;
+	}
+}
+
 int printf(const char* restrict format, ...) {
 	va_list parameters;
 	va_start(parameters, format);
-
+
```
Comment (AaronWantoch) [inline libc/stdio/printf.c:134]: ?

File: libc/stdlib/itoa.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+char get_ascii(int number){
+	if(number < 10){
+		return number+48;
+	}else{
+		return number+55;
+	}
+}
+
+void split_to_numbers(unsigned int value, size_t * num_of_char, char* str, int base){
```
Comment (AaronWantoch) [inline libc/stdlib/itoa.c:14]: Can we stick to one convention of pointers (where do we put the *) 
```suggestion
void split_to_numbers(unsigned int value, size_t* num_of_char, char* str, int base){
```

File: libc/string/strlwr.c [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include <string.h>
+
+char* strlwr(char* str){
+	int len = strlen(str);
+	for(size_t index = 0; index < len; index++){
+		if(64 < str[index] && str[index] < 91){
```
Comment (AaronWantoch) [inline libc/string/strlwr.c:6]: I would prefer it like that (my personal opinion)
```suggestion
		if(str[index] < 64 && str[index] < 91){
```

File: libc/string/strupr.c [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#include <string.h>
+
+char* strupr(char* str){
+        int len = strlen(str);
+        for(size_t index = 0; index < len; index++){
+                if(96 < str[index] && str[index] < 123){
```
Comment (AaronWantoch) [inline libc/string/strupr.c:6]: same as above
```suggestion
                if(str[index] > 96 && str[index] < 123){
```

File: libc/stdlib/itoa.c [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+
+char get_ascii(int number){
```
Comment (AaronWantoch) [inline libc/stdlib/itoa.c:6]: I don't understand how it works, if it is number from 0-9 we convert it into char equivalent. and number 11 means A right? What about small letters there is a gap in ASCI between big and small case letters. 
Please elaborate
image

**Review by AaronWantoch — COMMENTED** [review]
Some stylistic suggestions (no need to apply them). But I have one doubt.

File: libc/stdlib/itoa.c (hunk 5 above)
Comment (AaronWantoch) [inline libc/stdlib/itoa.c:6]: Can you elaborate on that function
I understand that number of value 0-9 will be converted into '0'-'9' and number of value 10 means 'A' and so on.
But what about small case letters? They start from value 97 and there is a gap between big case letters and small case letters.
image
If on the other hand we consider those signs (marked in yellow) what about signs with ASCI value 58-64.
From what I can see this function is unable to return them.
image

File: libc/stdlib/itoa.c (hunk 2 above)
Comment (AaronWantoch) [inline libc/stdlib/itoa.c]: Please stick to one convention of pointers (where we put '*' sign)
```suggestion
void split_to_numbers(unsigned int value, size_t* num_of_char, char* str, int base){
```

File: libc/string/strlwr.c (hunk 3 above)
Comment (AaronWantoch) [inline libc/string/strlwr.c:6]: I prefer it this way although it is my personal opinion
```suggestion
		if(str[index] > 64 && str[index] < 91){
```

File: libc/string/strupr.c (hunk 4 above)
Comment (AaronWantoch) [inline libc/string/strupr.c:6]: same as above
```suggestion
                if(str[index] > 96 && str[index] < 123){
```

**Review by AaronWantoch — CHANGES_REQUESTED** [review]
Stylistic changes may be ignored but please elaborate on get_ascii function

**Review by AaronWantoch — APPROVED** [review]

**Pull request merged**